Sources/NIOHTTP1/HTTPDecoder.swift

```diff
@@ -379,7 +379,12 @@ public class HTTPDecoder<HTTPMessageT>: ByteToMessageDecoder, AnyHTTPDecoder {
 
     public func decoderRemoved(ctx: ChannelHandlerContext) {
         // Remove the stored reference to ChannelHandlerContext
-        parser.data = UnsafeMutableRawPointer(bitPattern: 0x0000deadbeef0000)
+        #if arch(arm) // 32-bit
+            let veryDeadBeef : UInt = 0xdeadbeef
+            parser.data = UnsafeMutableRawPointer(bitPattern: veryDeadBeef)
+        #else
+            parser.data = UnsafeMutableRawPointer(bitPattern: 0x0000deadbeef0000)
+        #endif
```

Comment:
@jw Do we need this change? I think we only wanted this pointer value to be easy to spot in debuggers and core dumps: does it need to have the `deadbeef` portion shifted left 4 bits on 64-bit operating systems or is it ok to leave it in the least significant 32 bits?

Reply:
we can just use a 32 bit value `0xdeadbeef` is totally fine by me

Reply:
That still needs the UInt32 cast though, because parser.data seems to be Int32.

Reply:
@helje5 isn’t parser.data a pointer?

Reply:
Right it is. I guess the problem is the Swift type overloading again. It will treat 0xdeadbeef as an `Int`. At least that's the error w/o the explicit UInt (cannot represent Integer literal as Int or something like that).

Reply:
Sure, but can't we just compress this into one line? `parser.data = UnsafeMutableRawPointer(bitPattern: UInt(0xdeadbeef)`? That'll work correctly on all platforms.

Reply:
@helje5 I'm happy with this except for this